webpack/assets/javascripts/react_app/components/Editor/EditorReducer.js

```diff
@@ -0,0 +1,51 @@
+import Immutable from 'seamless-immutable';
+
+import {
+  EDITOR_CHANGE_STATE,
+  EDITOR_IMPORT_FILE,
+  EDITOR_REVERT_CHANGES,
+  EDITOR_EXEC_PREVIEW,
+} from './EditorConstants';
+
+const initialState = Immutable({
+  value: '',
+  renderedValue: '',
+  mode: 'Ruby',
+  theme: 'Monokai',
+  keyBinding: 'Default',
+  selectedView: 'input',
+  editorName: 'editor',
+  isMaximized: false,
+  isMasked: false,
+  isRendering: false,
+  readOnly: false,
+});
+
+export default (state = initialState, action) => {
+  const { payload } = action;
+
+  switch (action.type) {
+    case EDITOR_CHANGE_STATE: {
+      const [key] = Object.keys(payload);
+      if (Object.prototype.hasOwnProperty.call(state, key)) {
```

Comment:
how about moving this logic into the action (why firing an action if it doesn't change the state?) , plus throw an error if it doesn't exist.

Reply:
Great suggestion #2